src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessorClientFactoryOptions.cs

```diff
@@ -5,18 +5,18 @@
 
 namespace Microsoft.Health.Events.EventHubProcessor
 {
-    public class EventHubOptions
+    public class EventProcessorClientFactoryOptions
     {
-        public const string Settings = "EventHub";
+        public string EventHubNamespaceFQDN { get; set; }
 
-        public EventHubOptions(string connectionString, string name)
-        {
-            EventHubConnectionString = connectionString;
-            EventHubName = name;
-        }
-
-        public string EventHubConnectionString { get; set; }
+        public string EventHubConsumerGroup { get; set; }
 
         public string EventHubName { get; set; }
+
+        public string ConnectionString { get; set; }
```

Comment:
Rather than overloading these 2 options objects with both authentication methods, could we create separate implementations, create a common interface (IEventProcessorClientFactory), and just change which one gets added to the Service Collection at startup? We could just default to ConnectionString in the project, and allow consumers to just change it if needed.

e.g. EventProcessorClientConnectionStringFactoryOptions is injected into EventProcessorClientConnectionStringFactory.

There would be an implementation for Managed Identity - EventProcessorClientManagedIdentityFactoryOptions is injected into EventProcessorClientManagedIdentityFactory.

Both implement IEventProcessorClientFactory.

Reply:
I tried this out and like the idea of separating in principal, but the problem seems to be what methods IEventProcessorClientFactory should have. We want to be able to pass in options into the factories but if the options can either be EventProcessorClientConnectionStringFactoryOptions or EventProcessorClientManagedIdentityFactoryOptions then I think that means that IEventProcessorClientFactory would have to have methods that accept those options, or some sort of generic options (but then we have to check/validate the generic options that are being passed in are connection options or MI options which seems pretty similar to what the code is currently doing).